### DISCUSSION TITLE
How to remove rendered HTML from git while still being easily accessible

### DISCUSSION THREAD

**Discussion by termontwouter** [w3c/WebID#63] [General]
Based on @TallTed's https://github.com/w3c/WebID/pull/60#pullrequestreview-1872990063:

> HTML file really needs to be made non-git-managed, i.e., it should exist only as an artifact produced by CI operating on the BS file!

I agree. I initially put it there for ease of those who are used to having the rendered version available, but we can change that. The problem with artefacts is that there is not an easy way to link to those. I think using GitHub Pages to publish the latest rendering of each branch to would be handier.

Other ideas?

**Comment by TallTed** [comment]
GitHub Pages is a fine destination for CI-produced artefacts akin to PR-Preview. My focus is doing what we can to avoid people reviewing and suggesting changes to the HTML, when that HTML will be overwritten by output of BS processing, which will then lack the suggested changes. @iherman, @pchampin, or @msporny might be able to provide some guidance (or at least pointers to documentation and/or other helpful people) on how to get this rolling properly.

**Comment by jacoscaz** [comment]
@TallTed @termontwouter have a look at https://github.com/w3c/WebID/pull/62 .

**Comment by jacoscaz** [comment]
I've tried a few different strategies to keep everything within the same branch while working on #62 but ultimately kept stumbling into git's fundamental inability to ignore changes to tracked files.

The only way forward that I can see is to have the CI build maintain separate `${origin}-pages` branches, one per each branch that we commit to, and only push changes to build artifacts in these "special" branches.

**Reply by termontwouter** [reply]
I'm not entirely sure what you have tried, but I think it should be pretty easy to push the html files of each PR to different subdirectories of a single gh-pages branch... I've been rather busy lately, but should some time free up, I can do a proposal.

**Comment by jacoscaz** [comment]
I _might_ have addressed this in #70 , though it's the third time I find myself thinking of potentially having done so. Approach with skepticism.
